specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,176 @@
+# Background
+
+To improve the developer experience for customizing non-client regions, WebView2 is 
+working to support using DOM elements as non-client regions. We currently have limited 
+support for title bar aka draggable regions and are working on building out support 
+for caption controls and resize regions. 
+
+For security and flexibility, we want developers to be able to enable or disable all 
+custom non-client functionality per WebView. Non-client functionality will affect that 
+app window’s size and position so it’s important that developers can definitively 
+toggle access. This can be achieved in a limited way using a feature flag, but feature 
+flags are applied per WebView2 environment, thus, an API on the WebView2 to enable/disable 
+non-client support via a setting is the better solution.
+
+# Description
+`IsNonClientRegionSupportEnabled` defaults to `FALSE`. Disabling/Enabling 
+`IsNonClientRegionSupportEnabled` takes effect after the next navigation. Currently, draggable 
+regions is the only non-client experience we have implemented. Eventually, this setting will 
+expand to enable other non-client functionality, such as resize and caption controls. 
+
+When the setting is set to `TRUE`, then the following non-client region support for the top 
+level document will be enabled:  
+* Web pages will be able to use the `app-region` CSS style. 
+* Draggable Regions declared with the CSS style `app-region: drag` will support title bar 
+functionality, for example, the dragging of the app window, the title bar context menu upon 
+right click, and the maximizing/restoring of the window size upon double click of the html 
+element. 
+
+When set to `FALSE`, then all non-client region support will be disabled.  
+* Web pages will not be able to use the `app-region` CSS style. 
+
+# Examples
+This example enables non-client region support for all pages on www.microsoft.com. 
+Pages on other origins will not be trusted to use this feature. 
```

Comment:
This says 'Pages on other origins will not be trusted to use this feature.' - saying not trusting origins makes it sound like there's a potential security issue in allowing arbitrary web content to use this feature. Is that the case? If so, it needs to be noted in the ref docs for this property.

Reply:
We should change wording here to something more like "Pages on other origins will not have non-client region support enabled". There isn't a security issue, but a usability one if arbitrary pages are defining drag regions. It's not terrible, just annoying.

Reply:
updated